src/main/java/hudson/util/VersionNumber.java

```diff
@@ -513,16 +513,25 @@ public boolean isNewerThan(VersionNumber rhs) {
         return compareTo(rhs) > 0;
     }
 
+    /**
+     * Returns a digit (numeric component) by its position.
```

Comment:
This method is not doing what I would expect. You need to clarify what the `idx` refers to.

In the original method, `idx` only counts numeric segments so by my reading of the original code `new VersionNumber("1.0.0.GA.2-3").digit(3) == 2 && new VersionNumber("1.0.0.GA.2-3").digit(4) == undefined`

Your version, as I understand it, will have `new VersionNumber("1.0.0.GA.2-3").digit(3) == -1`

I wonder if the correct solution is to just deprecate this method and instead expose the items... or use a different parsing library